server/CGameHandler.cpp

```diff
@@ -93,29 +92,36 @@ class CBaseForGHApply
 template <typename T> class CApplyOnGH : public CBaseForGHApply
 {
 public:
-	bool applyOnGH(CGameHandler *gh, CConnection *c, void *pack, PlayerColor player) const override
+	bool applyOnGH(CGameHandler * gh, void * pack) const override
 	{
 		T *ptr = static_cast<T*>(pack);
-		ptr->c = c;
-		ptr->player = player;
-		return ptr->applyGh(gh);
+		try
+		{
+			return ptr->applyGh(gh);
+		}
+		catch(notAllowedActionException & e)
+		{
+			return false;
+		}
+		catch(...)
+		{
+			throw;
+		}
 	}
 };
 
 template <>
 class CApplyOnGH<CPack> : public CBaseForGHApply
 {
 public:
-	bool applyOnGH(CGameHandler *gh, CConnection *c, void *pack, PlayerColor player) const override
+	bool applyOnGH(CGameHandler * gh, void * pack) const override
```

Comment:
Use `static_assert` instead of `assert(0);`

Reply:
This one wouldn't work.